src/components/Navigation/index.jsx

```diff
@@ -9,26 +9,23 @@ class Navigation extends Component {
     platform: PropTypes.string.isRequired,
   };
 
-  constructor(props) {
-    super(props);
-    this.onChange = this.onChange.bind(this);
-  }
-
-  onChange(event) {
+  onChange = (event) => {
     const { name, value } = event.target;
-    const { history, platform } = this.props;
-    let newPlatform;
-    let newBenchmark;
+    const {
+      history, platform, benchmark, timeRange,
+    } = this.props;
+
+    let newPlatform = platform;
+    let newBenchmark = benchmark;
     if (name === 'platform') {
       newPlatform = value;
       newBenchmark = 'overview';
     } else {
-      newPlatform = platform;
       newBenchmark = value;
     }
     // eslint-disable-next-line react/prop-types
-    history.push(`/${newPlatform}/${newBenchmark}`);
-  }
+    history.push(`/${newPlatform}/${newBenchmark}?numDays=${timeRange}`);
```

Comment:
I see why you're passing `timerange` down here.
I think I'm noticing a new pattern.

I think we need a view/container component that we can use in the first Route to wrap the Navigation and DataControl components. That container component would also provide a function to Navigation and DataControl which they can then call to make history changes. In fact, such container component should hold a state of three values and the function should be in charge of changing the state and state changes will cause a history push.
We can handle it on a new issue to help you focus on completing this PR. Works for you?

Reply:
Yes, I agree, we definitely need that container component. In fact, I would like to work on it. As soon as I'm done here, I'll begin working on refactoring in the new issue. Thanks!

Reply:
Thinking more about this.
*I* (hand waving) introduced this problem by requesting two different components.
There's not a strong case for two components if it's going to make things a bit more complicated.
Should we go back to the only one component and not worry about this? I think so but I prefer to hear your thoughts.

Reply:
I think it depends upon the upcoming features. 

I read about these 2 other url updates - **Locking a data point** (append &dataPoint=<value> to URL) and **Toggle series** (append or remove &hide=<value> to URL) and I believe they should be placed in the same container. I couldn't find any other information regarding this topic. Please correct me if there should be more ways to update the path and the query strings.

Judging by its name, the Navigation component holds all the switches for the app's url, isn't that right? Being as few as they are, I don't think it would look too ugly to have all of them together. If it's ok with you, I can go ahead and remove the DataControl component and keep all the logic in Navigation.

Reply:
Let's keep it all under one (I believe that we're agreeing). Thanks @mihhu !